packages/feedback/src/index.ts

```diff
@@ -1 +1,375 @@
+import { createActorStyles } from './widget/Actor.css';
+import { Dialog } from './widget/Dialog';
+import { createDialogStyles } from './widget/Dialog.css';
+import { SuccessMessage } from './widget/SuccessMessage';
+
+export { sendFeedbackRequest };
+
+type ElectronProcess = { type?: string };
+
+// Electron renderers with nodeIntegration enabled are detected as Node.js so we specifically test for them
+function isElectronNodeRenderer(): boolean {
+  return typeof process !== 'undefined' && (process as ElectronProcess).type === 'renderer';
+}
+/**
+ * Returns true if we are in the browser.
+ */
+function isBrowser(): boolean {
+  // eslint-disable-next-line no-restricted-globals
+  return typeof window !== 'undefined' && (!isNodeEnv() || isElectronNodeRenderer());
+}
+
+const THEME = {
+  light: {
+    background: '#ffffff',
+    foreground: '#2B2233',
+    success: '#268d75',
+    error: '#df3338',
+  },
+  dark: {
+    background: '#29232f',
+    foreground: '#EBE6EF',
+    success: '#2da98c',
+    error: '#f55459',
+  },
+};
+
+/**
+ *
+ */
+export class Feedback implements Integration {
+  /**
+   * @inheritDoc
+   */
+  public static id: string = 'Feedback';
+
+  /**
+   * @inheritDoc
+   */
+  public name: string;
+
+  /**
+   * Feedback configuration options
+   */
+  public options: FeedbackConfigurationWithDefaults;
+
+  /**
+   * Reference to widget actor element (button that opens dialog).
+   */
+  private _actor: ReturnType<typeof Actor> | null;
+  /**
+   * Reference to dialog element
+   */
+  private _dialog: ReturnType<typeof Dialog> | null;
+  /**
+   * Reference to the host element where widget is inserted
+   */
+  private _host: HTMLDivElement | null;
+  /**
+   * Refernce to Shadow DOM root
+   */
+  private _shadow: ShadowRoot | null;
+  /**
+   * State property to track if dialog is currently open
+   */
+  private _isDialogOpen: boolean;
+
+  /**
+   * Tracks if dialog has ever been opened at least one time
+   */
+  private _hasDialogOpened: boolean;
+
+  public constructor({
+    showEmail = true,
+    showName = true,
+    useSentryUser = {
+      email: 'email',
+      name: 'username',
+    },
+    isAnonymous = true,
+    isEmailRequired = false,
+    isNameRequired = false,
+
+    buttonLabel = 'Report a Bug',
+    cancelButtonLabel = 'Cancel',
+    submitButtonLabel = 'Send Bug Report',
+    formTitle = 'Report a Bug',
+    emailPlaceholder = 'your.email@example.org',
+    emailLabel = 'Email',
+    messagePlaceholder = "What's the bug? What did you expect?",
+    messageLabel = 'Description',
+    namePlaceholder = 'Your Name',
+    nameLabel = 'Name',
+    successMessageText = 'Thank you for your report!',
+  }: Partial<FeedbackConfigurationWithDefaults> = {}) {
+    // Initializations
+    this.name = Feedback.id;
+    this._actor = null;
+    this._dialog = null;
+    this._host = null;
+    this._shadow = null;
+    this._isDialogOpen = false;
+    this._hasDialogOpened = false;
```

Comment:
would it compile/compress better if these values were initialized up above the constructor?

```diff
- private _actor: ReturnType<typeof Actor> | null;
+ private _actor: ReturnType<typeof Actor> | null = null;
```

Reply:
There's a lint rule that actually forbids this!

Reply:
ack

Reply:
it's actually the other way around (which is why we have this lint rule), it is smaller this way, because `private _field = null` is converted into a function that is called in the constructor, adding a few more bytes :)